### PR TITLE
docs(openapi): Fixed API documentation for V2

### PR DESCRIPTION
**Pull request by AnuthaDev**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [x] Documentation
- [ ] CI/CD

## Description

Improved OpenAPI documentation for V2

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Issues Addressed

- Merchant Account Update: Path Parameters in definition says it is {account_id}  instead that should be {id}

- Merchant Profile List Changes: Instead `Profile - list`  make `Merchant Account - Profile List` , Path Parameters in definition says it is {account_id}  instead that should be {id}

- Profile Changes: In profile all routes has [profile_id] in the path, change the path to [id] and the explanation in path parameter also to [id]. For `Merchant Connector - list` in side-bar, change to `Profile - Connector Accounts list`. In the definition of the api
`List Merchant Connector Details for the business profile` instead
`List Connector Accounts for the profile`

- Connector Account Changes: Instead of `Merchant Connector Account` use the term `Connector Account`

- Api-Key Changes: In the path param and its definition we are using {key_id] instead use [id]

- Routing Changes: In path param and its definition we are using {routing_algorithm_id}  instead is {id] , these changes are in api-documentation

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: api-reference-v2/api-reference/api-key/api-key--retrieve.mdx [hunk 1]
```diff
@@ -1,3 +1,3 @@
 ---
-openapi: get /v2/api_keys/{key_id}
+openapi: get /v2/api_keys/{id}
```
Comment (SanchithHegde) [inline api-reference-v2/api-reference/api-key/api-key--retrieve.mdx:2]: Should we consider using hyphens here instead, to keep things consistent:

```suggestion
openapi: get /v2/api-keys/{id}
```

CC: @jarnura @Narayanbhat166

Comment (SanchithHegde) [reply api-reference-v2/api-reference/api-key/api-key--retrieve.mdx:2]: We'd be taking this up in a separate PR.

File: api-reference-v2/mint.json [hunk 2]
```diff
@@ -76,7 +76,7 @@
       ]
     },
     {
-      "group": "Merchant Connector Account",
+      "group": "Connector Account",
       "pages": [
         "api-reference/merchant-connector-account/merchant-connector--create",
```
Comment (SanchithHegde) [inline api-reference-v2/mint.json:81]: Can we update the directory and file names to be "connector account" as well?

Comment (AnuthaDev) [reply api-reference-v2/mint.json:81]: Will create separate PR for this

Comment (AnuthaDev) [reply api-reference-v2/mint.json]: Done, please check

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**